qcodes/dataset/measurements.py

```diff
@@ -411,6 +414,8 @@ def reshaper(val, paramtype):
                 return float(val)
             elif paramtype == 'text':
                 return str(val)
+            elif paramtype == 'complex':
+                return complex(val)
```

Comment:
Even if this is perhaps a bit pathological it should have a test

Reply:
❤️